src/goto-instrument/nondet_static.cpp

```diff
@@ -31,7 +31,7 @@ void nondet_static(
 
   Forall_goto_program_instructions(i_it, init)
   {
-    const goto_programt::instructiont &instruction=*i_it;
+    const goto_programt::instructiont instruction=*i_it;
```

Comment:
Why take a copy?

Reply:
In case the instruction will be overriden, I need some info from the original instruction as it's being overriden (and without making a copy it gets into a loop). I could also make a copy of it only at point when I actually need it.